### PR TITLE
ToDo List project

### PR DESCRIPTION
**Pull request by cgcarlosg**
At this pull request I made a Todo project with all functionalities below:

Create todo with title, description, priority, complete check and, due date
Create projects to group Todos
Change icon-color depending on Todo priority
Remove when Todo is marked as complete
Delete todo
Use WebPack to create a bundle js file
Create WebPack configuration to bundle js file
Add scripts to package.json in order to run webpacks' build and watch

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,5 +1,287 @@
...
+
+function renderTaskCount(selectedList) {
+  const incompleteTaskCount = selectedList.tasks.filter(
+    (task) => !task.complete,
+  ).length;
+  const taskString = incompleteTaskCount === 1 ? 'task' : 'tasks';
+  listCountElement.innerText = `${incompleteTaskCount} ${taskString} remaining`;
+}
+
+function colorTasks(selectedList) {
```
Comment (Forison) [inline src/index.js]: * [ ] Let us use arrow function

File: src/index.js [hunk 2]
```diff
@@ -1,5 +1,287 @@
...
+    h2.textContent = 'New Task';
+    submitInput.value = 'Submit';
+    formContainer.style.pointerEvents = 'auto';
+    formContainer.style.transform = 'scale(1)';
+    overlay.style.opacity = 1;
+    modalOpen = true;
+  }
+}
+
+function openOrCloseUpdateTaskForm() {
```
Comment (Forison) [inline src/index.js]: * [ ] Let us use arrow function

File: src/index.js [hunk 3]
```diff
@@ -1,5 +1,287 @@
...
+    h2.textContent = 'Update Task';
+    submitInput.value = 'Update';
+    formContainer.style.pointerEvents = 'auto';
+    formContainer.style.transform = 'scale(1)';
+    overlay.style.opacity = 1;
+    modalOpen = true;
+  }
+}
+
+function renderTasks(selectedList) {
```
Comment (Forison) [inline src/index.js]: * [ ] Let us use arrow function

File: src/index.js [hunk 4]
```diff
@@ -1,5 +1,287 @@
...
+    editButton.classList.add('edit');
+    // eslint-disable-next-line no-use-before-define
+    editButton.addEventListener('click', () => editTask(task, label));
+    const todoTask = taskElement.querySelector('.task');
+    todoTask.append(editButton);
+    tasksContainer.appendChild(taskElement);
+  });
+}
+
+function render() {
```
Comment (Forison) [inline src/index.js]: * [ ] Let us use arrow function

File: src/index.js [hunk 5]
```diff
@@ -1,5 +1,287 @@
...
+    task.name = newTaskInput.value;
+    task.date = newTaskDate.value;
+    task.priority = newTaskPriority.value;
+    task.description = newTaskDescription.value;
+    label.innerHTML = `<span class="checkbox"></span>${task.name}<br>${task.date}<br>${task.description}`;
+    renderAndSave();
+  });
+}
+
+function createList() {
```
Comment (Forison) [inline src/index.js]: * [ ] Let us use arrow function

File: src/index.js [hunk 6]
```diff
@@ -1,5 +1,287 @@
...
+  return { id: Date.now().toString(), name: newListInput.value, tasks: [] };
+}
+
+function createTask() {
```
Comment (Forison) [inline src/index.js]: * [ ] Let us use arrow function and other areas when function keyword was used to define a function

**Review by Forison — CHANGES_REQUESTED** [review]
## Status::Changes Required❌

# 👋 Team

## You have done a good job on this one so far, I  like the UI, especially the modal. However, here are a few issues we have to address to get this approved💪.

* [ ] Do not commit unrelated files. Files like `DS_STORE` , `.VSCODE` should be remove.
* [ ] The link of the live link does not direct to the right path.
* [ ] User can delete a to-do from a given project.
* [ ] There is a 'default' project, even when no new project is added. For instance if no project is added any todo that is stored should be store under the default project.
* [ ] Let us add the necessary run common commands like `build`, `development` should be in `package.json` inside \`scripts'. Here is a basic example ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/scripts_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622602/))
should look like this,
```
javascript
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },
```
to make development more flexible.
* [ ] Let use arrow functions instead of the `function` keyword, check my comments below.

#### That is it, kindly make the change and submit a re-review😎.

#### Happy coding😇
Addo Forison

**Comment by cgcarlosg** [comment]
Hi. thanks for the feedback.

We have one question:

The button "Clear completed tasks" is the one used to remove or delete selected tasks.  Then we did not understand the comment about delete tasks.

Any additional comments please let us know.  Thanks

**Review by NtwaliHeritier — COMMENTED** [review]
### **STATUS: CHANGES REQUESTED🛑🛑**

Hello team,
The design is very attractive. Great work on making the requested changes. 
However one small change is required.

[REQUIRED]: Add the possibility to delete a task even when it is not completed.
[REQUIRED]: In your readme file, I think the correct way to run the project is to run npm run start instead of double clicking the index.html. Kindly modify it in the readme file.

That's it👍👍👍
Make the requested changes and ask for another review👌

**Review by arslanbisharat — APPROVED** [review]
# Approved :clap:

Hello Developer, wave:

Thank You for submitting this Pull Request. I am thrilled to announce that your project has met all the minimum requirements and now it is approved. :clap:

I am wishing you very good luck with the next project. :smile:

Once again, Congratulations!:joy:

**Best,
Arslan Bisharat(PT-TSE)
Slack: @arslanbisharat**

**Pull request merged**